File: src/test/java/io/github/welingtonmonteiro/multiplerun/ui/MultirunRunConfigurationEditorTest.java
package io.github.welingtonmonteiro.multiplerun.ui;

import static org.junit.Assert.assertEquals;

import java.util.Arrays;
import java.util.Collections;

import org.junit.Test;

public class MultirunRunConfigurationEditorTest {

    @Test
    public void addEnvProfilesAppendsNewPathsPreservingOrder() {
        assertEquals(Arrays.asList("/p/com.env", "/p/def.env", "/p/qa.env"),
                     MultirunRunConfigurationEditor.addEnvProfiles(
                             Collections.singletonList("/p/com.env"),
                             Arrays.asList("/p/def.env", "/p/qa.env")));
    }

    @Test
    public void addEnvProfilesSkipsDuplicatesAndBlanks() {
        assertEquals(Arrays.asList("/p/com.env", "/p/def.env"),
                     MultirunRunConfigurationEditor.addEnvProfiles(
                             Arrays.asList("/p/com.env", "/p/def.env"),
                             Arrays.asList("/p/com.env", "", "/p/def.env")));
    }

    @Test
    public void addEnvProfilesFromAnEmptyListKeepsEveryPicked() {
        assertEquals(Arrays.asList("/p/a.env", "/p/b.env"),
                     MultirunRunConfigurationEditor.addEnvProfiles(
                             Collections.emptyList(), Arrays.asList("/p/a.env", "/p/b.env")));
    }
}

File: src/test/kotlin/io/github/welingtonmonteiro/multiplerun/ui/MultirunRunConfigurationEditorTest.kt
package io.github.welingtonmonteiro.multiplerun.ui

import org.junit.Assert.assertEquals
import org.junit.Test

class MultirunRunConfigurationEditorTest {

    @Test
    fun addEnvProfilesAppendsNewPathsPreservingOrder() {
        assertEquals(listOf("/p/com.env", "/p/def.env", "/p/qa.env"),
                     MultirunRunConfigurationEditor.addEnvProfiles(
                         listOf("/p/com.env"),
                         listOf("/p/def.env", "/p/qa.env")))
    }

    @Test
    fun addEnvProfilesSkipsDuplicatesAndBlanks() {
        assertEquals(listOf("/p/com.env", "/p/def.env"),
                     MultirunRunConfigurationEditor.addEnvProfiles(
                         listOf("/p/com.env", "/p/def.env"),
                         listOf("/p/com.env", "", "/p/def.env")))
    }

    @Test
    fun addEnvProfilesFromAnEmptyListKeepsEveryPicked() {
        assertEquals(listOf("/p/a.env", "/p/b.env"),
                     MultirunRunConfigurationEditor.addEnvProfiles(
                         emptyList(), listOf("/p/a.env", "/p/b.env")))
    }
}
